core/src/main/java/io/paradoxical/cassieq/dataAccess/MessageRepositoryImpl.java

```diff
@@ -261,4 +265,26 @@ private void updateQueueSize(int amount){
 
         session.execute(with);
     }
+
+    /**
+     * Determines if the message is allowed to be consumed
+     * @param message
+     * @return
+     */
+    private boolean messageConsumable(final Message message) {
+        if(message.getDeliveryCount() >= queueDefinition.getMaxDeliveryCount()){
```

Comment:
We could have a separate class to manage consuming messages. with this in it that both the invoice and the reader use to mark them as consumed. Thoughts?